bigchaindb_driver/driver.py

```diff
@@ -1,57 +1,75 @@
+from abc import ABC, abstractproperty
 from bigchaindb_common.transaction import Transaction
 
 from .crypto import generate_keypair
 from .exceptions import InvalidVerifyingKey, InvalidSigningKey
 from .transport import Transport
 
 
-DEFAULT_NODE = 'http://localhost:9984/api/v1'
-
-
 class BigchainDB:
-    """BigchainDB driver class.
-
-    A :class:`~bigchaindb_driver.BigchainDB` driver is initialized with a
-    keypair and is able to create, sign, and submit transactions to a Node in
-    the Federation. At the moment, a BigchainDB driver instance is bounded to
-    a specific ``node`` in the Federation. In the future, a
-    :class:`~bigchaindb_driver.BigchainDB` driver instance might connect to
-    ``>1`` nodes.
+    """BigchainDB driver class connected to one or more BigchainDB nodes.
+
+    The BigchainDB API is exposed from :class:`~bigchaindb_driver.BigchainDB`
+    through a number of attribute namespaces:
+        - ``.transactions``: create, sign, and submit transactions
+
+    A custom class subclassing
+    :class:`~bigchaindb_driver.transport.AbstractTransport`'s interface can be
+    given to modify the behaviour of how to send requests (and where to, if
+    multiple nodes are given). By default, requests are sent to the given nodes
+    based on a simple round-robin algorithm (see
+    :class:`~bigchaindb_driver.pool.RoundRobinPicker`.
     """
     def __init__(self,
-                 *nodes,
+                 *node_urls,
                  verifying_key=None,
                  signing_key=None,
-                 transport_class=Transport):
+                 transport_cls=Transport,
+                 **kwargs):
         """Initialize a :class:`~bigchaindb_driver.BigchainDB` driver instance.
 
         If a :attr:`verifying_key` or :attr:`signing_key` are given, this
         instance will be bound to the keys and applied them as defaults
         whenever a verifying and/or signing key are needed.
 
         Args:
-            *nodes (str): BigchainDB nodes to connect to. Currently, the full
-                URL must be given. In the absence of any node, the default of
-                the :attr:`transport_class` will be used, e.g.:
+            *node_urls (str): BigchainDB nodes to connect to. Currently, the
```

Comment:
Right now every node connection is a url, but if this will change later, we can revert this.